August 2023 meeting minutes (#176)
onnx/sigs: operators/meetings/042-20230831.md

# Thursday, Aug 31st, 2023 at 9:00am PST

## Agenda

* Next release
* Deprecation of older opsets

## Attendees

* Aaron Bockover (Microsoft)
* Christian Bourjau (QuantCo)
* Scott Cyphers (Lightmatter)
* Cliff Houck
* Ganesan Ramalingam (Microsoft)
* Rajeev Rao (Nvidia)
* George (Intel)
* Jacky Chen (Microsoft)
* Justin Chu (Microsoft)
* Liqun Fu (Microsoft)
* Michal Karzynski (Intel)
* Xavier Dupré (Microsoft)

## Notes

### Next release

The next release (1.15) is planned to be around mid-October. The code-freeze for this
release will be late September. Liqun Fu (Microsoft) will be the release manager.
PRs targetting this release should use milestone 1.15

### Deprecation of older opsets

In the ONNX Steering Committee, Nvidia brought up the question of deprecation of older
opsets, arguing that it would be useful to deprecate older opsets, and have a policy for
doing so. The SIG meeting discussed this question.

First, there was a discussion about the earliest opset-versions that were supported by

> different frameworks/implementations and in use. Intel's openvino's earliest supported
> opset version is 7. The Pytorch exporter's base opset version is 9, while the new

Reviewer: ```suggestion
different frameworks/implementations and in use. Intel's OpenVINO's earliest supported
opset version is 7. The PyTorch exporter's base opset version is 9, while the new
```

upcoming Dynamo-based exporter targets opset version 18.

Subsequently, the discussion brought up the following questions
* What does it mean to deprecate an opset version?
* Why should we deprecate opset version?
* What problem are we trying to solve?

An initial suggestion was to follow a prevalent industry practice of first deprecating,
and then, after a period (like an year or so), marking the deprecated as obsolete,
and after another period, eventually removing it.

> This led to a deepper discussion of what impact deprecation has on the ONNX codebase? Will deprecated

Reviewer: ```suggestion
This led to a deeper discussion of what impact deprecation has on the ONNX codebase? Will deprecated
```

opset versions be removed from the ONNX codebase? There are a number of concerns with
doing this. The ONNX repo, being the gold-standard for the ONNX specification, should
maintain the definition of all opset versions for reference. Further, even with a shift
to newer opset versions, there will be a need for the version-converter to enable
conversion of older opset version models to newer opset versions, which will require

> the definition of older opset versions. Further-more, the way opset versions are defined

Reviewer: ```suggestion
the definition of older opset versions. Furthermore, the way opset versions are defined
```
